pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -479,6 +521,20 @@ public GenericRow getRecord(int docId, GenericRow reuse) {
           .getValue(docId, fieldSpec, _indexReaderWriterMap.get(column), _dictionaryMap.get(column),
               _maxNumValuesMap.getOrDefault(column, 0)));
     }
+
+    // Explicitly set it to null in case of reuse
+    reuse.putField(NULL_FIELDS, null);
+
+    // Look up bitmap for this docId and construct a comma separated
+    // null column names based on that bitmap
+    RoaringBitmap bitmap = _docIdToNullColumnsMap.get(docId);
```

Comment:
why can't we rebuild the row using `_docIdToNullColumnsMap`? Because we need to call `isPresent()` for all columns?

Do we know the overheads on this data structure in memory?

Reply:
Yes - currently we need to call isPresent to check which columns have corresponding null values for this docId.

Yes there's considerable overhead using a BitMap but not as much as using a comma separated string / Set

Reply:
I think if we move to a BitSet based approach (as mentioned above) - it'll make this part much simpler. We can simply save the Bitset as is per docId and use that to recreate the row.

Reply:
FYI: I pushed a change to optimize this part. Please take a look.